### PR TITLE
Add Deep Sound Channel calculated variable

### PR DESCRIPTION
**Pull request by htmlboss**
## Background
tldr: Someone wanted sound channel calculations, and so we deliver.

Available in:
* GIOPS Forecast 3D (Lat/Lon)
* GIOPS Forecast 3D PS
* RIOPS Forecast 3D PS

Deep Sound Channel is defined as the depth where the minimum sound speed occurs in a given vertical column.

I also added a basic test for the new `deepsoundchannel` function.

## Why did you take this approach?
Fastest performing implementation since it's vectorized.

## Anything in particular that should be highlighted?
Not really. Vectorized code sometimes isn't the most intuitive to understand so I've added some comments to hopefully clarify things.

## Screenshot(s)
Screenshot from 2020-02-11 10-16-39

Screenshot from 2020-02-11 11-17-30

## Bonus
Fixed a crash when calculating pressure with GIOPS Forecast 3D (Lat/Lon) so speed of sound is now available there too.

## Checks
- [x] I ran unit tests.
- [x] I've tested the relevant changes from a user POV.

_Hint_ To run all python unit tests run the following command from the root repository directory:
```sh
python -m unittest `find tests -name "*.py" | grep -v disabled`
```

### CODE REVIEW

**Comment by htmlboss** [comment]
Ignore codefactor complaining...numpy doesn't play well with `isinstance`.
